lib/git-shell-out-strategy.js

```diff
@@ -723,6 +739,16 @@ export default class GitShellOutStrategy {
     return this.exec(args, {useGitPromptServer: true, writeOperation: true});
   }
 
+  /**
+   * Undo Operations
+   */
+  reset(type, revision = 'HEAD') {
+    const validTypes = ['soft'];
```

Comment:
No `'hard'` or `'mixed'` yet? :smile:

Reply:
heh nope, not yet. @annthurium and I (and maybe @simurai?) talked about it a bit and decided to hold off on those until we have more thorough discussion about UX and other implications. For example, if we implement a way to do a hard reset then we probably want to add some safeguards so that users don't accidentally blow away workdir changes in a way that is unrecoverable. And if someone accidentally does a mixed reset after carefully staging changes, then we've just blown away their index state and we'd probably want to offer a way for them to get back to the state they were in before the reset. 

All of this to say, I'd like to have more developed thoughts on the recovery story if users accidentally do one of these operations. Or alternatively, decide on a UX that makes it really hard for users to accidentally or incorrectly do one of these operations. What do you think @smashwilson 💭